tests/suite_test.go

```diff
@@ -0,0 +1,191 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/anthhub/forwarder"
+	"github.com/go-logr/logr"
+	mfc "github.com/manifestival/controller-runtime-client"
+	mf "github.com/manifestival/manifestival"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	dspav1alpha1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1alpha1"
+	systemsTesttUtil "github.com/opendatahub-io/data-science-pipelines-operator/tests/util"
+	"go.uber.org/zap/zapcore"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"testing"
+	"time"
+)
+
+const (
+	APIServerPort = 8888
+)
+
+var (
+	loggr     logr.Logger
+	ctx       context.Context
+	cfg       *rest.Config
+	cancel    context.CancelFunc
+	clientmgr ClientManager
+
+	APIServerURL         string
+	kubeconfig           string
+	k8sApiServerHost     string
+	DSPAPath             string
+	DSPANamespace        string
+	skipDeploy           bool
+	skipCleanup          bool
+	PortforwardLocalPort int
+	DSPA                 *dspav1alpha1.DataSciencePipelinesApplication
+	forwarderResult      *forwarder.Result
+)
+
+var (
+	DefaultDeployTimeout time.Duration
+	DefaultPollInterval  time.Duration
+	DefaultDeleteTimeout time.Duration
+)
+
+type ClientManager struct {
+	k8sClient client.Client
+	mfsClient mf.Client
+	mfopts    mf.Option
+}
+
+// TestAPIs - This is the entry point for Ginkgo -
+// the go test runner will run this function when you run go test or ginkgo.
+// Under the hood, Ginkgo is simply calling go test.
+// You can run go test instead of the ginkgo CLI, But Ginkgo has several capabilities that can only be accessed via ginkgo.
+// It is best practice to embrace the ginkgo CLI and treat it as a first-class member of the testing toolchain.
+func TestAPIs(t *testing.T) {
+	// Single line of glue code connecting Ginkgo to Gomega
+	// Inform our matcher library (Gomega) which function to call (Ginkgo's Fail) in the event a failure is detected.
+	RegisterFailHandler(Fail)
+
+	// Inform Ginkgo to start the test suite, passing it the *testing.T instance and a description of the suite.
+	// Only call RunSpecs once and let Ginkgo worry about calling *testing.T for us.
+	RunSpecs(t, "Controller Suite")
+}
+
+// Register flags in an init function. This ensures they are registered _before_ `go test` calls flag.Parse()
+func init() {
+	flag.StringVar(&kubeconfig, "kubeconfig", "~/.kube/config", "The path to the kubeconfig.")
+	flag.StringVar(&k8sApiServerHost, "k8sApiServerHost", "localhost:6443", "The k8s cluster api server host.")
```

Comment:
should we extract these magic values used for param defaults to a `const` in the head of this file?